realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -385,6 +390,11 @@ private static RealmCache getCache(String realmPath, boolean createIfNotExist) {
             throw new IllegalArgumentException(ASYNC_CALLBACK_NULL_MSG);
         }
 
+        // If there is no Realm file it means that we need to sync the initial remote data in the worker thread.
+        if (!configuration.realmExists()) {
```

Comment:
This should also check if it's a sync configuration. If not then it will never be removed again.